### DISCUSSION TITLE
Cannot find symbol import com.google.firebase.FirebaseApp;

### DISCUSSION THREAD

**Discussion by miloandme** [invertase/react-native-firebase#4506] [Q&A] [answered]
Hello, In our react-native project, we are currently using

"react": "16.8.3",
"react-native": "0.59.9",
"react-native-firebase": "5.5.4",

I have upgraded the modules to

"react": "16.13.1",
"react-native": "0.63.3",
"@react-native-firebase/app": "8.4.7",
"@react-native-firebase/database": "^7.5.13"

but I am getting an error on native android side, since 
we have implemented native android code also using Firebase.

Firebase can be accessed on native android before from this version of the library "react-native-firebase": "5.5.4",
but now I am not getting "Cannot resolve symbol FirebaseApp"

How can Firebase be accessed on native android using this new version "@react-native-firebase/app": "8.4.7" ?

I tried to unlink and link manually, but I did not work.
Probably React Native 0.60+ have auto linking

These are the screenshots.

![Firebase1](https://user-images.githubusercontent.com/73989958/98232919-3aaff700-1f99-11eb-9931-7aeea4dbc887.png)
![Firebase2](https://user-images.githubusercontent.com/73989958/98232924-3c79ba80-1f99-11eb-9cd5-794e803c0347.png)

**Comment by mikehardy** [comment, marked as answer]
Hi there! This is not a react-native-firebase problem, it's just a generic java code problem for Android.
The symbols you want are defined by firebase-android-sdk and if you are accessing them natively (outside the context of react-native-firebase access in javascript), then you need to follow the native instructions from the upstream library. Here's their example for accessing the database natively: https://github.com/firebase/quickstart-android/tree/master/database

**Reply by miloandme** [reply]
Thank you.
